WordImpress/Give-Addon-Boilerplate
# Refactor: update starter setup

## Description



This PR fixes broken setups and adds new ones. 

The following was implemented:

- Support to TypeScript
- Settings page using React App 
- JS helper function to retrieve images from `public/images` folder
- Support to PHPUnit tests + sample tests
- GitHub workflows
- Fix `build.php` and add support to the new files added to the project

## Affects



The starting point to create new add-ons. Now will be quicker to start new projects. 

## Visuals



![image](https://user-images.githubusercontent.com/16308864/235149881-b6bdc4b0-3785-40bc-b4c3-01f014f52541.png)

![image](https://user-images.githubusercontent.com/16308864/235150004-710b1247-b649-445c-8f9f-fec31b31efd6.png)

## Testing Instructions



1. Clone this repository to your local
2. Remove the .git directory
3. Run php `build.php` from the CLI
4. Run `composer install` from the CLI
5. Run `npm install` from the CLI
6. Run `npm run dev` from the CLI to compile the assets
7. Run `composer test` from the CLI to run the PHPUnit test
8. Go to `WP > Plugins` admin page and activate it
9. Go to `Give > Settings` to check the new pages added there

## Pre-review Checklist
